hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -165,8 +165,12 @@ protected void setPendingInflightAndRequestedInstants(Set<String> pendingInfligh
   protected void preCommit(HoodieCommitMetadata metadata) {
     // Create a Hoodie table after startTxn which encapsulated the commits and files visible.
     // Important to create this after the lock to ensure the latest commits show up in the timeline without need for reload
-    HoodieTable table = createTable(config, hadoopConf);
-    resolveWriteConflict(table, metadata, this.pendingInflightAndRequestedInstants);
+    HoodieTable<?, I, ?, T> table = createTable(config, hadoopConf);
+    try {
```

Comment:
NTR: Even though HoodieTable implements AutoCloseable, I could not use try with resources construct. I wanna throw something specific if close of HoodieTable fails vs something fails within try block. hence had to take this route.